test/test.js

```diff
@@ -70,12 +70,34 @@ describe('typing-effect', function () {
   describe('delay attributes', function () {
     it('uses defaults when no delays specified', function () {
       const typingEffectElement = document.createElement('typing-effect')
+      typingEffectElement.setAttribute('data-reduced-motion', false)
       document.body.append(typingEffectElement)
 
       assert.equal(typingEffectElement.characterDelay, 40)
       assert.equal(typingEffectElement.lineDelay, 40)
     })
   })
+
+  describe('a11y considerations', function () {
+    it('sets delay to 0 when media query matches (prefers-reduced-motion)', function () {
+      const typingEffectElement = document.createElement('typing-effect')
+      document.body.append(typingEffectElement)
+
+      assert.equal(window.matchMedia('(prefers-reduced-motion)').matches, true)
+      assert.equal(typingEffectElement.characterDelay, 0)
+      assert.equal(typingEffectElement.lineDelay, 0)
+    })
```

Comment:
I'm surprised that we didn't need to set `prefers-reduced-motion` as part of the setup for this test, since I would imagine the default would be for `prefers-reduced-motion` not to be set, which would cause `characterDelay` and `lineDelay` to be `40`.

@koddsson: do you know if there's a good way to set `prefers-reduced-motion` in our tests?

Reply:
I tried something like the following:
```
global.window = {
  matchMedia: (str) => str === '(prefers-reduced-motion)' ? true : false
} 
```

but got yelled at by the compiler 😭

Reply:
I pushed 3d6d644, which mocks the API. The compiler didn't complain to me.

Reply:
Hmm, not sure why it was giving me grief then. Maybe because of the `global` declaration. Thanks for the assist!

Reply:
Leave a comment